redbot/cogs/mod/names.py

```diff
@@ -69,6 +69,88 @@ class ModInfo(MixinMeta):
             else:
                 await ctx.send(_("Done."))
 
+    def handle_custom(self, user):
+        a = [c for c in user.activities if c.type == discord.ActivityType.custom]
+        if not a:
+            return None, discord.ActivityType.custom
+        a = a[0]
+        c_status = None
+        if not a.name and not a.emoji:
+            return None, discord.ActivityType.custom
+        if not a.name:
+            c_status = self.bot.get_emoji(a.emoji.id)
+        if c_status:
+            pass
+        if a.name and a.emoji:
+            c_status = f"{a.emoji} {a.name}"
+        elif a.emoji and not c_status:
+            c_status = f"{a.emoji}"
+        elif a.name:
+            c_status = a.name
+        else:
+            c_status = None
+        return c_status, discord.ActivityType.custom
+
+    def handle_playing(self, user):
+        p_acts = [c for c in user.activities if c.type == discord.ActivityType.playing]
+        p_act = p_acts[0] if p_acts else None
+        act = p_act.name if p_act and p_act.name else None
+        return act, discord.ActivityType.playing
+
+    def handle_streaming(self, user):
```

Comment:
It would be nice if this would also show the platform user streams on (`activity.platform` - it can be Twitch or YouTube).
It might be difficult to show it as "Streaming on Platform" if you don't choose to take the suggestion about returning proper string from `handle_*` functions which I put in review comment on L142-151.